core/cloudflow-operator/src/main/scala/cloudflow/operator/runner/SparkRunner.scala

```diff
@@ -194,6 +197,141 @@ object SparkRunner extends Runner[CR] with PatchProvider[SpecPatch] {
   private val OnFailureRetryIntervalSecs           = 10
   private val OnSubmissionFailureRetryIntervalSecs = 60
 
+  private def addDriverResourceRequirements(driver: Driver, podsConfig: PodsConfig, deployment: StreamletDeployment)(
+      implicit ctx: DeploymentContext
+  ): Driver = {
+    var updatedDriver = driver
+    import ctx.sparkRunnerSettings._
+
+    // this logic means you can only set cores to an int. and it blows up for decimals
+    // From Spark Operator (which is not supported yet):
+    //
+    //   CoreRequest is the physical CPU core request for the driver.
+    //   Maps to `spark.kubernetes.driver.request.cores` that is available since Spark 3.0.
+    //
+    val coresOpt = toIntCores(driverSettings.cores)
+
+    updatedDriver = updatedDriver.copy(
+      cores = coresOpt,
+      memory = driverSettings.memory.map(_.value),
+      coreLimit = driverSettings.coreLimit.map(_.value),
+      memoryOverhead = driverSettings.memoryOverhead.map(_.value)
+    )
+    // you can set the "driver" pod or just "pod", which means it will be used for both driver and executor (as fallback).
+    updatedDriver = podsConfig.pods
+      .get(DriverPod)
+      .orElse(podsConfig.pods.get(PodsConfig.CloudflowPodName))
+      .flatMap { podConfig =>
+        podConfig.containers.get(PodsConfig.CloudflowContainerName).flatMap { containerConfig =>
+          containerConfig.resources.map { resources =>
+            updatedDriver.copy(
+              cores = toIntCores(resources.requests.get(Resource.cpu)).orElse(coresOpt),
+              memory = resources.requests.get(Resource.memory).map(_.toString).orElse(updatedDriver.memory),
+              coreLimit = resources.limits.get(Resource.cpu).map(_.toString).orElse(updatedDriver.coreLimit)
+            )
+          }
+        }
+      }
```

Comment:
So if the user just sets generic pod settings, that will override specific settings for the Spark driver? Isn't it more sensible to be the other way around (i.e. more specific settings like driver-level ones take precedence over higher level settings)?

Reply:
the updatedDriver is already set with defaults, and here the cores, memory and limit are always set, taking the values from the podConfig first falling back to existing settings (from defaults), so  the precedence is as you would like it.